NMS-8704: EIF Adapter

UberPinguin opened the pull request:
Add an IBM Tivoli Event Integration Facility adapter to OpenNMS.
The EIF adapter accepts events sent using the EIF protocol from IBM Tivoli systems including ITM and Netcool/OMNIbus, and parses them into OpenNMS events.

RangerRick commented:
I recommend we squash-merge this when the time comes, since there's a lot of refactor/merge stuff in the commit stream.

j-white commented on features/eif-adapter/pom.xml:
```diff
@@ -0,0 +1,112 @@
+  <parent>
+    <groupId>org.opennms</groupId>
+    <artifactId>org.opennms.features</artifactId>
+    <version>19.0.0-SNAPSHOT</version>
```
Develop is 20.0.0, so you'll need to bump up the version here.

j-white commented on features/eif-adapter/src/test/java/org/opennms/features/eifadapter/EifParserTest.java:
```diff
@@ -0,0 +1,16 @@
+package org.opennms.features.eifadapter;
```
Copyright header.

j-white commented on opennms-base-assembly/src/main/filtered/etc/org.opennms.features.eif-adapter.cfg:
```diff
@@ -0,0 +1,4 @@
+# Interface on which to listen
```
We can omit including this in the base assembly and rely on the Blueprint's defaults instead.

UberPinguin replied:
OK. I'll need to update the admin doc, too.

j-white commented on opennms-doc/guide-admin/src/asciidoc/text/events/eif-adapter.adoc:
```diff
@@ -0,0 +1,53 @@
+
+[[ga-events-eif-configuring]]
+==== Configuring
+First check that your _OpenNMS_ installation supports this feature.
```
We'll need to update this section now that the config file is gone.

You can assume that if this content is in the documentation for the corresponding version, then the feature will be present.

UberPinguin replied:
Argh. I had updated it, but it didn't get added to the commit. Revision coming in a moment.

j-white commented on opennms-doc/guide-admin/src/asciidoc/text/events/eif-adapter.adoc:
```diff
@@ -0,0 +1,53 @@
+First check that your _OpenNMS_ installation supports this feature.
+If it does there should be a `${OPENNMS_HOME}/etc/org.opennms.features.eif-adapter.cfg` file.
+
+Now open the file, review its content and make sure to apply the correct settings depending on your environment.
```
I prefer to use the Karaf shell commands to udpate the .cfg files instead of editing this directly. See
`http://docs.opennms.org/opennms/releases/19.0.1/guide-install/guide-install.html#gi-install-minion-configure` for an example.

UberPinguin replied:
Neat! Is there a way to activate the config changes without restarting all of OpenNMS?

j-white replied:
Running `config:update` will apply the changes immediately since `update-strategy="reload"` is set in the Blueprint.

UberPinguin commented:
Updated the documentation to use Karaf console instead of a config file.
Removed references to the config file.

There's currently an issue with installing directly using `feature:install eif-adapter`, as Karaf is unable to resolve all of the dependency constraints automatically. I don't _think_ this is a bug in the EIF feature configuration, but if I'm wrong I will correct it.

UberPinguin commented on opennms-doc/guide-admin/src/asciidoc/text/events/eif-adapter.adoc:
```diff
@@ -0,0 +1,49 @@
+[root@localhost /root]# $ ssh -p 8201 admin@localhost
+...
+opennms> features:install eif-adapter
+opennms> config:edit org.opennms.features.eifadapter
```
In practice, I had to install `opennms-collection-api` before Karaf would install the `eif-adapter` feature.
I _think_ this is a bug elsewhere, so I did not include it in the documentation here.
`eif-adapter` depends on `opennms-dao-api`, which depends on `opennms-config-api`, which can't resolve a requirement for `org.opennms.netmgt.collection.adapters` until I manually install `opennms-collection-api`, which .. depends on `opennms-config-api` ?

UberPinguin commented on opennms-doc/guide-admin/src/asciidoc/text/events/eif-adapter.adoc:
```diff
@@ -0,0 +1,49 @@
+----
+[root@localhost /root]# $ ssh -p 8201 admin@localhost
+...
+opennms> features:install eif-adapter
```
In practice, I had to install `opennms-collection-api` before Karaf would install the `eif-adapter` feature.
I _think_ this is a bug elsewhere, so I did not include it in the documentation here.
`eif-adapter` depends on `opennms-dao-api`, which depends on `opennms-config-api`, which can't resolve a requirement for `org.opennms.netmgt.collection.adapters` until I manually install `opennms-collection-api`, which .. depends on `opennms-config-api` ?

j-white commented:
Squash merged to `develop`
